Assets/Scripts/Utils/StSCamera.cs

```diff
@@ -1,12 +1,175 @@
+using System;
 using System.Collections;
 using System.Collections.Generic;
+using Cinemachine;
 using UnityEngine;
+using UnityEngine.InputSystem;
 using Utils;
 
 /// <summary>
 /// A basic class that just ensures we only have one camera loaded at a time. 
 /// </summary>
 public class StsCamera : Singleton<StsCamera>
 {
+    [Serializable]
+    public enum CameraStateName
+    {
+        Default,
+        SelectedStagePosition,
+    }
+    
+    [Serializable]
+    public struct CameraState
+    {
+        public CameraStateName Name;
+        [Tooltip("This can be left null, stage positions will automatically fill this.")]
+        public Transform focusTarget; 
+        public float fieldOfView;
+        public float blendDuration;
+    }
+    [Header("Camera States")][Space(20)]
+    [SerializeField] private List<CameraState> cameraStates;
+    [SerializeField] private CameraState currentCameraState;
+    [SerializeField] private CinemachineVirtualCamera vCam;
+    private CinemachineFramingTransposer vCamTransposer;
+    
+    [Header("Camera Input")]
+    [SerializeField] private bool IsMovingCamera = false;
+    private PlayerInput input;
+    private InputAction onPointerPress;
+    private InputAction onPointerDelta;
+     private Vector2 panDelta;
+    [SerializeField] private Vector2 panMin;
+    [SerializeField] private Vector2 panMax;
+    [SerializeField] private float panSensitvity;
+    [Header("The amount of time it takes for panning to reset to the target when changing focus.")]
+    [SerializeField] private float panResetDuration = 1;
+    
+    private void OnEnable()
+    {
+        input = FindObjectOfType<PlayerInput>();
```

Comment:
Obviously we only want these actions handled or even bound to during chapter play but we can deal with that later